### ISSUE TITLE
loads images inside container with display: none;

### ISSUE THREAD

**Issue by Lewiscowles1986** [aFarkas/lazysizes#376]
I thought adding display: none; would stop it from lazy loading images. Unfortunately this is not happening and images are still loaded. Is this expected behaviour?

**Comment by Lewiscowles1986** [comment]
**Update:**

```
if(lazyloadElems[i].offsetParent === null){continue;}
```

Testing the following code right now, seems to be working. Thanks [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent)

**Comment by aFarkas** [comment]
Thank you, but I can not reproduce your issue. See here for a testcase: https://jsbin.com/jixadehefe/edit?html,console,output

Can you please produce a testcase where the current check fails?

**Comment by Lewiscowles1986** [comment]
Current iteration of https://jujubedev.mybigcommerce.com although the code above fixes and will be in during the week. 

Interestingly we're manually checking the resources panel and with google pagespeed insights because data-src="" will not be loaded by the browser without the JS.

What we're trying to do is essentially stop images loading if the element is hidden directly or via parent.

**Comment by aFarkas** [comment]
This isn't a test case. It is a full page. Most other projects do never accept such bug reports by principal. I often do otherwise, but it would be very helpful, if you could simply say which image is hidden and downloaded anyways, otherwise I need to check the hole site to only see the behavior.

Basically a bug report has to include all steps to reproduce a bug. And a bug report that states "somewhere on the following page is the bug" doesn't help much. If you check my [jsbin](https://jsbin.com/negewozufu/edit?html,console,output) you see that I already put more work into reproducing your bug than you.

**Comment by Lewiscowles1986** [comment]
The entire theme suffers the bug without the single-line patch.

You asked if I had a code example. I simply provided one along with two very simple ways to verify the images are loaded by the code and not the browser (which doesn't recognise data-src attribute as a place to get images from). 

It has one place it includes the lazysizes code, and I've a line in above comments that patches for my situation.

If you believe it is necessary or compatible with your library, or anyone in the future wants to save themselves time and effort by using, it's published publicly with no license, so can be used anywhere. 

I'll leave decisions about your library to you as I frankly don't have the time or inclination. 

I don't believe looking at your test, that it is testing what you think I am reporting, or that the issue has been understood, but that's fine and I'm sorry for any confusion or frustration I have caused. 

This report is as much for others using lazysizes as it is for you as an author, and I'd feel rude not raising an issue being that you've done a lot of the effort of creating a lazy-loading library used in a project I'm working on, included by the original theme author.
